tests/test_encoding.py

```diff
@@ -39,17 +39,17 @@ def test_bom(self):
         utf32le = b"\xff\xfe\x00\x00\x34\x6c\x00\x00"
         for string in (utf16be, utf16le, utf32be, utf32le):
             bom_encoding, bom = read_bom(string)
-            assert bom_encoding is not None
-            assert bom is not None
+            self.assertIsNotNone(bom_encoding)
+            self.assertIsNotNone(bom)
```

Comment:
This change is breaking mypy, because mypy does not understand self.assert*.

If flake8 complains about asserts, I believe we need to disable that check for all files (we will also use asserts outside tests for mypy’s sake).

I am a bit puzzled because in Scrapy we use asserts often, but I see nothing about ignoring such an error in https://github.com/scrapy/scrapy/blob/master/.flake8 (the ignored errors seem unrelated).

Reply:
I made that change to avoid a bandit issue. It was something like:
```
>> Issue: [B101:assert_used] Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   Location: tests/test_url.py:1063:8
   More Info: https://bandit.readthedocs.io/en/0.0.0/plugins/b101_assert_used.html
```

Reply:
[Oh, there it is indeed!](https://github.com/scrapy/scrapy/blob/1c9d308accd38a91ffa92e3aff8912cd792070eb/.bandit.yml#L2)

Reply:
https://github.com/scrapy/scrapy/blob/1c9d308accd38a91ffa92e3aff8912cd792070eb/.bandit.yml#L21 would also be useful in any repo.

Reply:
Done in https://github.com/scrapy/w3lib/pull/220/commits/5390b23153070145d167135412a354eafc5ab95c